src/benchmark.rs

```diff
@@ -18,37 +18,39 @@ impl Miner {
             "Benchmarking. This will take {} sec...",
             TEST_DURATION
         ));
+        // inside the rayon threadpool the tokio runtime is not available.
+        // we access the runtime where it is available (outside of the pool) and spawn tasks directly on that runtime.
+        let rt = tokio::runtime::Handle::current();
         let handles: Vec<_> = (0..args.threads)
+            .into_par_iter()
             .map(|i| {
-                std::thread::spawn({
-                    move || {
-                        let timer = Instant::now();
-                        let first_nonce = u64::MAX.saturating_div(args.threads).saturating_mul(i);
-                        let mut nonce = first_nonce;
-                        loop {
-                            // Create hash
-                            let _hx = drillx::hash(&challenge, &nonce.to_le_bytes());
-
-                            // Increment nonce
-                            nonce += 1;
-
-                            // Exit if time has elapsed
-                            if (timer.elapsed().as_secs() as i64).ge(&TEST_DURATION) {
-                                break;
-                            }
-                        }
+                rt.spawn(async move {
```

Comment:
Running CPU intensive tasks in tokio runtime is highly discouraged. You should use `task::spwan_blocking` instead. 

https://docs.rs/tokio/latest/tokio/task/index.html#blocking-and-yielding

Reply:
[`d894bb4` (#103)](https://github.com/regolith-labs/ore-cli/pull/103/commits/d894bb433a3648ff123c8fe32b3057523a387d0f)